canonical/ubuntu.com
# fix(dynamic-forms.js): Don't remove name attribute from form inputs after processing

## Done

- There was some logic in dynmaic-forms.js that was pretty loosely removing the `name` attribute from inputs. This was fine for checkboxes and such, but bad for name, email, hidden inputs, etc. This removes this functionality. _Instead_, we should not be adding the `name` attribute to fields we don't want submitted to the payload.
- As a flyby I removed the `validateForm()` function. It was being ran on three forms only, here is what I did with them:
**hpc**: removed the .js-form-validate class as there were no required fields
**kernel-beta-signup**: deleted the form as it was not used anywhere and the beta is over
**intel-iot**: refactored into a single page, so the `validateForm()` function was not needed.

### Fly-by

- in `marketo_submit` function initially assign and object to enrichment_fields. `enrichment_fields = {}`. Before it was being assigned `None`, making it useless in the later steps.

## QA

- Check the form/payload is being properly submitted for:
https://ubuntu-com-14244.demos.haus/download/iot/intel-iot#get-in-touch
https://ubuntu-com-14244.demos.haus/hpc#get-in-touch
- also check the forms here:
https://ubuntu-com-14244.demos.haus/landscape/pricing
As this is where the issue was originally reported

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-14580
https://warthogs.atlassian.net/browse/WD-14270
https://github.com/canonical/ubuntu.com/issues/14194
